src/start.S

```diff
@@ -14,18 +14,22 @@ _start:
 .option norelax
     la gp, _global_pointer
 .option pop
-    la sp, _stack_end
     csrw sstatus, zero
     csrw sie, zero
 
-    la t1, kernel_init
-    la ra, 1f
-    jr t1
-
-    j 1f
+    // Clear the BSS
+    la   a3, _bss_start
+    la   a4, _bss_end
 1:
+    sd   zero, (a3)
+    addi a3, a3, 8
+    blt  a3, a4, 1b
+
+    la   sp, _stack_end
+    call kernel_init
+2:
     wfi
-    j 1b
+    j 2b
```

Comment:
2b on line 49?

>     la t1, secondary_init
>     la ra, 1b
>     jr t1

Reply:
Whoops, good catch! Ended up just giving the wfi loop it's own label to make this more clear.